### ISSUE TITLE
Error running with eBPF on Kernel >= 5.8.5

### ISSUE THREAD

**Issue by kris-nova** [falcosecurity/falco#1463]
**Describe the bug**

Running Falco with BPF on kernels 8.5 and 9 result in runtime error

**How to reproduce it**

Compile Falco 0.26.1

```
git clone git@github.com:falcosecurity/falco.git
cd falco
git checkout tags/0.26.1 -b tag-0.26.1
mkdir build
cd build
cmake ../ \
      -DBUILD_BPF="ON" \
      -DBUILD_WARNINGS_AS_ERRORS="OFF" \
      -DCMAKE_BUILD_TYPE="Release" \
      -DCMAKE_INSTALL_PREFIX="/usr" \
      -DFALCO_ETC_DIR="/etc/falco" \
      -DUSE_BUNDLED_DEPS=ON
make bpf
make falco
sudo cp driver/bpf/probe.o /root/.falco/falco-bpf.o
sudo cp userspace/falco/falco /usr/bin/falco
```

Run Falco with eBPF

```
sudo FALCO_BPF_PROBE="" falco
```

Runtime Error

```
1178: (25) if r1 > 0x80 goto pc+679
 R0=inv(id=0) R1_w=inv(id=0,umax_value=128,var_off=(0x0; 0xff)) R2_w=inv(id=0) R3_w=map_value(id=0,off=0,ks=4,vs=1612,imm=0) R5_w=inv(id=0) R6_w=ctx(id=0,off=0,imm=0) R7_w=map_value(id=0,off=0,ks=4,vs=77,imm=0) R8=inv(id=0) R9=inv4294967294 R10=fp0 fp-72=mmmmmmmm fp-80=map_value fp-88=map_value fp-96=map_value fp-104=map_value fp-112=map_value fp-120=ctx fp-128=map_value fp-136=mmmmmmmm fp-144=mmmmmmmm fp-152=map_value fp-160=inv16 fp-168=mmmmmmmm fp-176=map_value fp-184=mmmmmmmm fp-192=mmmmmmmm
1179: (79) r1 = *(u64 *)(r10 -96)
1180: (bf) r3 = r8
1181: (85) call bpf_probe_read#4
R2 min value is negative, either use unsigned or 'var &= const'
processed 2228 insns (limit 1000000) max_states_per_insn 1 total_states 93 peak_states 93 mark_read 39
Wed Oct 28 10:40:39 2020: Runtime error: bpf_load_program() err=13 event=filler/sys_read_x message=0: (bf) r6 = r1
1: (b7) r1 = 0
2: (63) *(u32 *)(r10 -72) = r1
last_idx 2 first_idx 0
regs=2 stack=0 before 1: (b7) r1 = 0
3: (bf) r2 = r10
4: (07) r2 += -72
5: (18) r1 = 0xffff8e787d0b4400
7: (85) . Exiting.
```

**Expected behaviour**

**Screenshots**

**Environment**

- Falco version: 0.26.1

- System info:  

```json
{
  "machine": "x86_64",
  "nodename": "emily",
  "release": "5.9.1-arch1-1",
  "sysname": "Linux",
  "version": "#1 SMP PREEMPT Sat, 17 Oct 2020 13:30:37 +0000"
}

```

- Cloud provider or hardware configuration:
- OS: Arch

- Kernel: Linux emily 5.9.1-arch1-1

- Installation method: Compile from source

**Additional context**

**Comment by fntlnz** [comment]
I’ve been looking into this for a long time.

It is a bit more complex than just the kerne being >= 5.8.5 and also happens in older kernel with different kernel/clang combinations.

Let’s keep the discussion on the original one where the works is happening
https://github.com/falcosecurity/falco/issues/1251

**Comment by kris-nova** [comment]
I was able to fix this by using `clang version 9.0.1` 

see  https://github.com/falcosecurity/falco/issues/1251#issuecomment-718324102 for instructions on how I was able to fix this
